### ISSUE TITLE
TypeScript paths is ignored

### ISSUE THREAD

**Issue by TimMensch** [remoteinterview/zero#121]
In my tsconfig I use the following pattern:

```jsonc
{
        "paths": {
            /* The ~ option for easier imports: "~some/path" */
            "~*": ["./*"]
        },
}
```

This means I can specify imports as follows:

```
import { Layout } from "~components/Layout";
```

This pattern eliminates the annoying `"../.../components/Layout"` style relative paths that need to be divined based on what folder you're currently in, or adjusted as you move files or copy code around. VS Code also supports this for auto-imports. It's relatively standard, and is also supported by `parcel`. 

Zero, unfortunately, sees `"~components"` as a _package_ and tries (and fails) to import it. A simple fix would be to grab the `paths` entry from `tsconfig.json` and ignore any imports that match any of the path globs.

**Comment by asadm** [comment]
I agree about this particular shorthand pattern ie. `~` for project root. The underlying parcel also supports it. I would count this as a bug in zero, I will look into this soon. 

About parsing `tsconfig.json`.`paths` and ignoring those paths, I think it would be a good enhancement. If we just address the common patterns (like the `~` usage) this might be just unnecessary effort.

**Comment by TimMensch** [comment]
It's true that `~` usage will hit the 99th percentile of usefulness. If that's a lot easier, then I'd say it's fine. Was thinking that just checking against the globs would work, but I could be wrong.

**Comment by asadm** [comment]
It looks like there is an underlying [parcel bug too](https://github.com/parcel-bundler/parcel/issues/2813). I guess I will have to patch that and see if that works. I will try to get this fix up this weekend.

**Comment by asadm** [comment]
@TimMensch can you try the installing the latest version. #122 fixes the root imports using ‘~’

**Comment by TimMensch** [comment]
Thanks for the fast fix! Seems to work great. Closing.

I was testing `zero` for my site, and it's wonderful for what it does, but I ended up running into a different limitation (or at least a problem I couldn't solve) that made me try Next.js instead -- and now I have a working Next.js environment. I can't remember what the limitation was, though it may have been that I couldn't figure out how to get it to spit out valid AMP HTML? Next.js has native AMP support. There are specific things that AMP sites need, like a `<style amp-custom>` block, that React and similar don't support well, and I wanted to _also_ use MDX in the same site for content.

The current project is more important to _finish_ than to use the perfect platform. I'll absolutely consider `zero` for my next project, though.

Thanks again!
